Make modular for the use in other states for logfile watching

joshboon opened the pull request:
Hey folks,

I've modded this to allow other states to generate beaver config and have beaver automatically include it. I've specifically used it to catch my apache error logs across instances and ship them. Let me know if anyone would find this useful. It also still handles the old config as expected in my testing so shouldn't have an impact on anyone else. Sample for the modular route follows:
/etc/beaver/conf.d/vhost-{{ vhost['servername'] }}.conf:
  file:
    - managed
    - source: salt://apache-vhosts/beaver-error.conf
    - user: root
    - group: root
    - mode: 0644
    - template: jinja
    - backup: minion
    - require: 
      - file: /etc/beaver/conf.d
    - context:
      servername: {{ vhost['servername'] }}
      apacheenv: {{ pillar['apacheenv'] }}
cat apache-vhost/beaver-error.conf 
[/var/www/vhosts/{{servername}}/statistics/logs/error_log]
type: apache-error
tags: {{apacheenv}},{{servername}}

joshboon committed:
mod for modular logfiles

joshboon committed:
fix include dir creation and logfiles check

joshboon committed:
fix dir include

joshboon committed:
remove logfiles ref

joshboon committed:
adjust log include again

joshboon committed:
fix logfiles check

joshboon committed:
move conf.d to onchanges

joshboon committed:
move to monitoring full beaver dir

joshboon committed:
add more file reqs to get service to start

joshboon committed:
adjust file check for logs

joshboon committed:
better check for logfiles

joshboon committed:
add systemd support

joshboon committed:
add systemd install support

aboe76 commented:
is it intentional to remove the redhat specific init file? and keep the debian init file?

joshboon commented:
It's moved to installing a systemd version which is found on almost all recent linux OS's. Note I don't remove the file so if a person were to run this on a systemd enabled system they would also have the old init version too which could cause issues.

joshboon committed:
Update map.jinja

Deal with Py3

joshboon committed:
Update map.jinja

definitely not an e

joshboon committed:
Update map.jinja

beaver is sadly not py3 compat

joshboon committed:
Update map.jinja

updated pip command

myii commented:
Hey, @joshboon -- it seems you've been waiting a _short_ while for some responses here!  This formula looks pretty unmaintained.  Has this been replaced by the following formula?

* https://github.com/saltstack-formulas/logstash-formula

If not, maybe we could find a different solution to make it easier for you to contribute here.

joshboon commented:
I'd actually recommend it be deprecated in favor of filebeat https://github.com/saltstack-formulas/filebeat-formula/ as beaver was meant to be run as a logshipper on a remote server that wouldn't want to run logstash or couldn't run logstash as it doesn't have enough memory.

myii commented:
> I'd actually recommend it be deprecated in favor of filebeat https://github.com/saltstack-formulas/filebeat-formula/ as beaver was meant to be run as a logshipper on a remote server that wouldn't want to run logstash or couldn't run logstash as it doesn't have enough memory.

@joshboon That makes sense.  How would you like to proceed?  Is there anything from this PR that could be useful there?  Even if we do decide to archive here, your contribution here would be worth adding to this archive, in the event anyone ever comes back to this formula.

joshboon committed:
revert to previous config for python2 native

joshboon committed:
add archive message

joshboon commented:
@myii I've updated to add a warning this requires python2 and reverted my attempts to make it compatible in a mixed environment. I chose not to immediately send folks to filebeat as they aren't drop-in replacements for each other. I think this would make the most sense for the current situation.

myii commented:
Thanks for your work here, @joshboon.  So this repo is ready to be archived in your opinion?

joshboon commented:
Yup beaver needs a maintainer too so it'd cause more harm than good to use this instead of a more recent log shipper.

myii commented:
> Yup beaver needs a maintainer too so it'd cause more harm than good to use this instead of a more recent log shipper.

Thanks, will do so shortly.
